rust/xaynet-analytics/src/database/analytics_event/data_model.rs

```diff
@@ -39,30 +54,37 @@ impl<'screen> AnalyticsEvent<'screen> {
     }
 }
 
-impl<'screen> IsarAdapter for AnalyticsEvent<'screen> {
-    fn into_field_properties() -> IntoIter<FieldProperty> {
-        // NOTE: properties need to be ordered by type. Properties with the same type need to be ordered alphabetically
-        // https://github.com/isar/isar-core/blob/1ea9f27edfd6e3708daa47ac6a17995b628f31a6/src/schema/collection_schema.rs
-        vec![
-            FieldProperty::new("event_type".to_string(), DataType::Int),
-            FieldProperty::new("name".to_string(), DataType::String),
-            FieldProperty::new("screen_route".to_string(), DataType::String),
-            FieldProperty::new("timestamp".to_string(), DataType::String),
-        ]
-        .into_iter()
-    }
+impl TryFrom<AnalyticsEventRelationalAdapter> for AnalyticsEvent {
+    type Error = anyhow::Error;
 
-    fn write_with_object_builder(&self, object_builder: &mut ObjectBuilder) {
-        object_builder.write_int(self.event_type as i32);
-        object_builder.write_string(Some(&self.name));
-        object_builder.write_string(self.screen_route.map(|screen| screen.name.as_ref()));
-        object_builder.write_string(Some(&self.timestamp.to_rfc3339()));
+    fn try_from(adapter: AnalyticsEventRelationalAdapter) -> Result<Self, Self::Error> {
+        let event = AnalyticsEvent::new(
+            adapter.name,
+            TryInto::<AnalyticsEventType>::try_into(adapter.event_type)
```

Comment:
This can be `adapter.event_type.try_into()` or if it complain about the type
 `adapter.event_type.try_into::<AnalyticsEventType>()`

Reply:
done